### PR TITLE
Ref #1533 - Set location view text field to scheme-less domain url

### PR DESCRIPTION
**Pull request by chickdan**
## Summary of Changes

This pull request fixes issue #1533 

This change sets the `urlTextField` using the `domainURL`. Setting it here allows for the full url to still be accessible when the user taps into the location view.

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Screenshots:

Shortened domainURL
Simulator Screen Shot - iPhone Xs - 2019-10-10 at 23 49 05

Full url when location view tapped
Simulator Screen Shot - iPhone Xs - 2019-10-10 at 23 49 12

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — DISMISSED** [review]
Thanks for the contribution, the code works well :) 
I'm adding `blocked` label, waiting on final decision from the product team

File: Client/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift
Comment (iccub) [inline Client/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift]: This `domainURL` strips `m.` and `mobile.` schemes as well.
I talked to security team and they want to hide scheme only with `www.` case.

So one additional helper method needs to be added to `URLExtensions.swift` to remove only the `www` part

**Review by iccub — CHANGES_REQUESTED** [review]

**Comment by chickdan** [comment]
@iccub I've pushed an update for the requested change. Please let me know if you have any suggestions for improvement.

File: Shared/Extensions/URLExtensions.swift
Comment (iccub) [inline Shared/Extensions/URLExtensions.swift]: All of this code is duplicated and the only difference is between `"^(www)\\."`, `"^(www|mobile|m)\\."` regexes
What do you think about some DRY refactor?

- change `domainURL` to a function `func domainURL(stripWWWSubdomainOnly: Bool = false) -> URL`
- same for `normalizedHost`
- in `normalizedHost` add something like `let regex = stripWWWSubdomainOnly ? "^(www)\\." : "^(www|mobile|m)\\."`
- update documentation comment in `domainURL`
- update `testdomainURL` unit test

**Review by iccub — CHANGES_REQUESTED** [review]

File: Shared/Extensions/URLExtensions.swift
Comment (chickdan) [reply Shared/Extensions/URLExtensions.swift]: Great suggestions! I've made the requested changes and added a new unit test to account for the `stripWWWSubdomainOnly` parameter.

**Review by iccub — APPROVED** [review]
Great stuff, thanks :)

**Comment by iccub** [comment]
Just needs a rebase and we are good

**Comment by chickdan** [comment]
conflicts resolved :)

**Comment by iccub** [comment]
CI still failing with 
```
/Users/travis/build/brave/brave-ios/BraveShared/Analytics/UrpService.swift:28:44: cannot call value of non-function type 'URL?'
        guard let hostUrl = try? host.asURL(), let normalizedHost = hostUrl.normalizedHost() else { return nil }
```

**Pull request merged**
